kbs/src/api/src/policy_engine/opa/mod.rs

```diff
@@ -51,105 +33,211 @@ impl PolicyEngineInterface for Opa {
         &self,
         resource_path: String,
         input_claims: String,
-    ) -> Result<(bool, String)> {
-        let policy = tokio::fs::read_to_string(
-            self.policy_path
-                .to_str()
-                .ok_or_else(|| anyhow!("Missing Policy Path"))?,
-        )
-        .await
-        .map_err(|e| anyhow!("Reading OPA policy file failed: {:?}", e))?;
-
-        let policy_go = GoString {
-            p: policy.as_ptr() as *const c_char,
-            n: policy.len() as isize,
-        };
-
-        let resource_path_input = serde_json::json!({ "resource-path": resource_path }).to_string();
-
-        let resource_path_go = GoString {
-            p: resource_path_input.as_ptr() as *const c_char,
-            n: resource_path_input.len() as isize,
-        };
-
-        let input_go = GoString {
-            p: input_claims.as_ptr() as *const c_char,
-            n: input_claims.len() as isize,
-        };
-
-        // Call the function exported by cgo and process the returned decision
-        let decision_buf: *mut c_char =
-            unsafe { evaluateGo(policy_go, resource_path_go, input_go) };
-        let decision_str: &CStr = unsafe { CStr::from_ptr(decision_buf) };
-        let res = decision_str.to_str()?.to_string();
-        log::debug!("Evaluated: {}", res);
-        if res.starts_with("Error::") {
-            return Err(anyhow!(res));
-        }
-
-        let res_kv: Value = serde_json::from_str(&res)?;
-        let result_boolean = res_kv["allow"]
-            .as_bool()
-            .ok_or_else(|| anyhow!("Policy Engine output must contain \"allow\" boolean value"))?;
-
-        Ok((result_boolean, res))
+    ) -> Result<bool, ResourcePolicyError> {
+        let mut engine = regorus::Engine::new();
+
+        // Add policy as data
+        engine
+            .add_policy_from_file(self.policy_path.clone())
+            .map_err(|_| ResourcePolicyError::PolicyLoadError)?;
+
+        // Add resource path as data
+        let resource_path_object =
+            regorus::Value::from_json_str(&format!("{{\"resource-path\":\"{}\"}}", resource_path))
+                .map_err(|_| ResourcePolicyError::ResourcePathError)?;
```

Comment:
same concern es with the other errors, we discard the information carried in the original error, which might be useful for debugging.